More ssh agent information (#409)
SUSE/doc-caasp: adoc/deployment-preparation.adoc

In order to deploy {productname} you need a workstation running {sle} 15 SP1 or similar {opensuse} equivalent.
This workstation is called the "Management node".

[[ssh.configuration]]
=== Basic SSH key configuration

To log into the created cluster nodes, you need to configure an SSH key pair and
load it into your users `ssh-agent` program. This is also mandatory in order to be able to use
the installation tools `terraform` and `skuba`. In a later deployment step,
skuba will ensure that the key is distributed across all the nodes and trusted by them.
For now, you only need to make sure that an ssh-agent is running and that it has
the SSH key added:

. The `ssh-agent` is usually started automatically by graphical
desktop environments. If that is not your case run:
+
----
eval "$(ssh-agent)"
----
This will start the agent and set environment variables used for agent
communication within the current session. This has to be the same terminal session
that you run the `skuba` commands in. A new terminal usually requires a new ssh-agent.

> . In some desktop environments the ssh-agent will also automatically load the SSH keys.

Author: Remove extra period / space
```suggestion
In some desktop environments the ssh-agent will also automatically load the SSH keys.
```

> To add an SSH keys manually, use the `ssh-add` command:

Author: singular "key"
```suggestion
To add an SSH key manually, use the `ssh-add` command:
```

+
----
ssh-add <path_to_key>
----
+
[TIP]
====

> If you are adding the SHH key manually, specify the full path.

Author: ```suggestion
If you are adding the SSH key manually, specify the full path.
```

For example: `/home/sles/.ssh/id_rsa`
====

You can load multiple keys into your agent using the `ssh-add <path_to_key>` command.

> Keys should be password protected for security measures. The

Author: ```suggestion
Keys should be password protected as a security measure. The
```

ssh-add command will prompt for your password, then the agent caches the
private key for a configurable lifetime. The `-t lifetime` option to
ssh-add specifies a maximum time to cache the key. See `man ssh-add` for
more information.

[NOTE]
====
Skuba will try all the identities loaded into the ssh-agent until one of
them grants access to the node, or until the ssh server maximum
authentication attempts are exhausted.
====

==== Forwarding the authentication agent connection
It is also possible to *forward the authentication agent connection* from a
host to another one, which can be useful if you intend to run skuba on
a "jump host" and don't want to copy your private key to this node.
This can be achieved using the `ssh -A` command. Please refer to the man page

> of SSH to learn about the security implications of using this feature.

Author: This is referring to the command `ssh`, not the protocol "SSH", so lower-case might be more accurate.  Maybe?  I'm somewhat indifferent.

Reviewer: usually man pages are referred to in lowercase, or even more specifically `ssh(1)`.
